docs/source/json-api/production-setup.rst

```diff
@@ -0,0 +1,300 @@
+set a custom root CA certificate used to validate client certificates via ``--cacrt ca.crt``
+
+
+Architecture
+************
+
+Components
+----------
+
+A production setup of the *HTTP-JSON API* will involve the following components:
+
+- the *HTTP-JSON API* server
+- the query store backend database server
+- the ledger
+
+*HTTP-JSON API* server exposes an API to interact with the Ledger and it uses JDBC to interact
+with its underlying query store for caching and serving data efficiently.
+
+The *HTTP-JSON API* server releases are regularly tested with OpenJDK 8 on a 64-bit x86 architecture,
+with Ubuntu 20.04, macOS 11.5.2 and Windows Server 2016.
+
+In production, we recommend running on a 64-bit x86 architecture in a Linux
+environment. This environment should have a Java SE Runtime Environment such
+as OpenJDK JRE and must be compatible with OpenJDK version 1.8.0_202 or later.
+We recommend using PostgreSQL server as query-store, most of our tests have
+been done with servers running version > 10.
+
+
+Scaling and Redundancy
+**********************
+
+.. note:: This section of the document only talks about scaling and redundancy setup for the *HTTP-JSON API* server. In all of the recommendations suggested below we assume that the JSON API always interacts with a single participant on the ledger.
+
+We advise that the *HTTP-JSON API* server and *query store* components to have dedicated
+computation and memory resources available to them. This can be achieved via
+containerization or setting them up on independent physical servers. Ensure that the two
+components are **physically co-located** to reduce network latency for
+communication. The scaling and availability aspects heavily rely on the interactions between
+the core components listed above.
+
+With respect to scaling we recommend to follow the general advice in trying to
+understand the bottlenecks and see if adding additional processing power/memory
+i.e vertical scaling is beneficial.
+
+In general for horizontal scaling purposes , we recommend to treat the
+*HTTP-JSON API* and the *query store database server* as a single unit for scaling.
+While a setup with multiple *HTTP-JSON API* services running against a single backend
+database server is feasible, it maybe a futile exercise if the database server
+itself is the bottleneck.
+
+We can run a redundant setup for the *HTTP-JSON API* by using a reverse proxy server with some
+acceptable routing mechanism and then and having multiple *HTTP-JSON API* servers sit behind it
+each with their own backend database servers and dedicated computation and memory resources.
+
+Users may consider running PostgreSQL backend in a `high availability configuration <https://www.postgresql.org/docs/current/high-availability.html>`__.
+The benefits of this are use-case dependent as this may be more expensive for
+smaller active contract datasets, where re-initializing the cache is cheap and fast.
```

Comment:
This is the first time this document implies that it is acceptable to just drop the DB and start from scratch. I think this should be said explicitly, and much earlier, likely around line 23 (i.e. near the start of the "Query store" section).